src/inmanta/data/dataview.py

```diff
@@ -1211,19 +1211,21 @@ def get_base_url(self) -> str:
 
     def get_base_query(self) -> SimpleQueryBuilder:
         base = SimpleQueryBuilder(
-            select_clause="""SELECT a.name, a.environment, last_failover, paused, unpause_on_resume,
-                                            ap.hostname as process_name, ai.process as process_id,
-                                            (CASE WHEN paused THEN 'paused'
-                                                WHEN id_primary IS NOT NULL THEN 'up'
-                                                ELSE 'down'
+            select_clause="""SELECT a.name, a.environment, a.last_failover, a.paused, a.unpause_on_resume,
+                                            (CASE WHEN a.paused THEN 'paused'
+                                                WHEN NOT a.paused AND schedagent.id_primary IS NULL THEN 'down'
+                                                ELSE 'up'
                                             END) as status""",
-            from_clause=f" FROM {Agent.table_name()} as a LEFT JOIN public.agentinstance ai ON a.id_primary=ai.id "
-            " LEFT JOIN public.agentprocess ap ON ai.process = ap.sid",
-            filter_statements=[" a.environment = $1 "],
-            values=[self.environment.id],
+            from_clause=f" FROM  (SELECT sched.id, sched.halted FROM {Environment.table_name()} AS sched "
+            "WHERE sched.id =$1) AS sched "
+            f"RIGHT JOIN {Agent.table_name()} a ON a.environment = sched.id "
+            f"RIGHT JOIN (SELECT schedagent.name, schedagent.id_primary FROM {Agent.table_name()} AS schedagent "
```

Comment:
```suggestion
            from_clause= f"RIGHT JOIN (SELECT schedagent.name, schedagent.id_primary FROM {Agent.table_name()} AS schedagent "
```

I think only the last bit is used?

Reply:
Why a right join?